src/main/webapp/feedback/load-icon-feedback.js

```diff
@@ -0,0 +1,100 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import ParsedIconFeedback from './parsed-icon-feedback.js';
+
+export default class LoadIconFeedback {
+  static #ENDPOINT_FEEDBACK = '/icon-feedback';
+  static #PARAM_LECTURE_ID = 'lectureId';
+  static #INCREMENT_INTERVAL = 10000;
+
+  #lectureId;
+  #parsedIconFeedback;
+
+  constructor(lectureId) {
+    this.#lectureId = lectureId;
+  }
+
+  async initialize() {
+    this.#parsedIconFeedback = new ParsedIconFeedback();
+    await this.loadIconFeedbackList();
+  }
+
+  /**
+   * Fetches avaiable IconFeedback from `ENDPOINT_FEEDBACK`
+   * and parses the data for it to be graphed.
+   */
+  async loadIconFeedbackList() {
+    const url =
+        new URL(LoadIconFeedback.#ENDPOINT_FEEDBACK, window.location.origin);
+    url.searchParams.append(
+        LoadIconFeedback.#PARAM_LECTURE_ID, this.#lectureId);
+    const response = await fetch(url);
+    const jsonData = await response.json();
+    this.parseFeedback(jsonData);
+    console.log(this.#parsedIconFeedback);
+  }
+
+  /**
+   * Parses `iconFeedbackJson` by couting how many times each IconFeedback type
+   * is clicked in each 10 second interval and stores that data in a
+   * ParseIconFeedback object.
+   */
+  parseFeedback(iconFeedbackJson) {
+    let index = 0;
+    let interval = 0;
+    while (index < iconFeedbackJson.length) {
+      let goodCount = 0;
+      let badCount = 0;
+      let tooFastCount = 0;
+      let tooSlowCount = 0;
+      while (index < iconFeedbackJson.length &&
```

Comment:
Consider refactoring this to avoid nested loops and tracking the index:

```javascript
for (const iconFeedback of iconFeedbackJson) {
  if (timestamp < interval) {
    update interval, append/reset counts
  }
  ... increment count
}
append counts
```